content/guides/integrations/redis/Java.md

```diff
@@ -9,152 +9,167 @@ logo: /images/java-opencensus.png
-redis/roundtrip_latency|The latency spent for various Redis operations
-redis/reads|The number of reads performed
-redis/writes|The number of writes performed
-
-## End to end example
+### Available metrics
+
+Metric search suffix|Description|Tags
+---|---|---
+jedis/latency|The latency of the various methods|"method", "error", "status"
+jedis/calls|The calls made by various methods|"method", "error", "status"
+jedis/data_transferred|The amount of data transferred|"method", "type"
+
+
+### End to end example
+
+In this end-to-end example, we've taken an excerpt from a search service. It checks Redis server if the expensively computed response
+was already saved. If not found ("cache miss"), it'll process the response then save it to the cache for later "cache hits".
+
 {{<tabs Source Pom>}}
 
 {{<highlight java>}}
 package io.opencensus.tutorials.jedis;
 
-import io.opencensus.exporter.stats.stackdriver.StackdriverStatsConfiguration;
-import io.opencensus.exporter.stats.stackdriver.StackdriverStatsExporter;
-import io.opencensus.exporter.trace.stackdriver.StackdriverTraceConfiguration;
-import io.opencensus.exporter.trace.stackdriver.StackdriverTraceExporter;
+import io.opencensus.trace.Tracer;
 import io.opencensus.trace.Tracing;
+import io.opencensus.trace.Span;
+import io.opencensus.trace.Status;
 import io.opencensus.trace.config.TraceConfig;
 import io.opencensus.trace.config.TraceParams;
 import io.opencensus.trace.samplers.Samplers;
-import java.io.BufferedReader;
-import java.io.InputStreamReader;
-import java.io.IOException;
+import io.opencensus.common.Scope;
+
+// Enable the exporters.
+import io.opencensus.exporter.stats.prometheus.PrometheusStatsCollector;
+import io.opencensus.exporter.trace.zipkin.ZipkinTraceExporter;
+import io.prometheus.client.exporter.HTTPServer;
 
-import redis.clients.jedis.Jedis;
-import redis.clients.jedis.Observability;
+import io.orijtech.integrations.ocjedis.OcWrapJedis;
+import io.orijtech.integrations.ocjedis.Observability;
 
 public class JedisOpenCensus {
-    private static final Jedis jedis = new Jedis("localhost");
+    private static final OcWrapJedis jedis = new OcWrapJedis("localhost");
+    private static final Tracer tracer = Tracing.getTracer();
 
     public static void main(String ...args) {
         // Enable exporting of all the Jedis specific metrics and views.
         Observability.registerAllViews();
 
         // Now enable OpenCensus exporters
-        setupOpenCensusExporters();
-
-        // Now for the repl
-        BufferedReader stdin = new BufferedReader(new InputStreamReader(System.in));
+        try {
+            setupOpenCensusAndExporters();
+        } catch(Exception e) {
+            System.err.println("Failed to create OpenCensus exporters " + e);
+        }
 
         while (true) {
             try {
-                System.out.print("> ");
-                System.out.flush();
-                String query = stdin.readLine();
+                Thread.sleep(1700);
```

Comment:
Can you please add a comment here about 1700 sleep?

Reply:
Roger that, thanks!